library/src/interactor_impl.cxx

```diff
@@ -196,145 +196,55 @@ class interactor_impl::internals
-        break;
-      case 'N':
-        self->Interactor.triggerCommand("toggle ui.filename");
-        break;
-      case 'M':
-        self->Interactor.triggerCommand("toggle ui.metadata");
-        break;
-      case 'Z':
-        self->Interactor.triggerCommand("toggle_fps");
-        break;
-      case 'R':
-        self->Interactor.triggerCommand("toggle render.raytracing.enable");
-        break;
-      case 'D':
-        self->Interactor.triggerCommand("toggle render.raytracing.denoise");
-        break;
-      case 'V':
-        self->Interactor.triggerCommand("toggle_volume_rendering");
-        break;
-      case 'I':
-        self->Interactor.triggerCommand("toggle model.volume.inverse");
-        break;
-      case 'O':
-        self->Interactor.triggerCommand("toggle model.point_sprites.enable");
-        break;
-      case 'U':
-        self->Interactor.triggerCommand("toggle render.background.blur");
-        break;
-      case 'K':
-        self->Interactor.triggerCommand("toggle interactor.trackball");
-        break;
-      case 'F':
-        self->Interactor.triggerCommand("toggle render.hdri.ambient");
-        break;
-      case 'J':
-        self->Interactor.triggerCommand("toggle render.background.skybox");
-        break;
-      case 'L':
+      // Modifiers version not found, try ANY instead
+      bind.Modifiers = ModifierKeys::ANY;
+      commandsIt = self->InteractionCommands.find(bind);
+    }
+
+    if (commandsIt != self->InteractionCommands.end())
+    {
+      for (const std::string& command : commandsIt->second)
       {
-        std::string command =
-          rwi->GetShiftKey() ? "decrease_light_intensity" : "increase_light_intensity";
-        self->Interactor.triggerCommand(std::move(command));
-        break;
-      }
-      case 'H':
-        self->Interactor.triggerCommand("toggle ui.cheatsheet");
-        break;
-      case '?':
-        self->Interactor.triggerCommand("print_scene_info");
-        break;
-      case '1':
-        self->Interactor.triggerCommand("set_camera front");
-        break;
-      case '3':
-        self->Interactor.triggerCommand("set_camera right");
-        break;
-      case '4':
-        self->Interactor.triggerCommand("roll_camera -90");
-        break;
-      case '5':
-        self->Interactor.triggerCommand("toggle scene.camera.orthographic");
-        break;
-      case '6':
-        self->Interactor.triggerCommand("roll_camera 90");
-        break;
-      case '7':
-        self->Interactor.triggerCommand("set_camera top");
-        break;
-      case '9':
-        self->Interactor.triggerCommand("set_camera isometric");
-        break;
-      default:
-        if (keySym == F3D_EXIT_HOTKEY_SYM)
+        if (!self->Interactor.triggerCommand(command))
         {
-          self->Interactor.triggerCommand("stop_interactor");
+          log::error("Interaction: error running command:\"", command, "\", ignoring");
```

Comment:
is that needed? failure is already logged in `triggerCommand()`

Reply:
Not systematically, if the command is found but the command itself returns false (but does not log for some reason, who knows what libf3d users will do), this is the only log we will get.